### DISCUSSION TITLE
Improve sql storage

### DISCUSSION THREAD

**Discussion by flexsurfer** [status-im/status-legacy#16621] [General]
| Table                                       | Rows  | 
|---------------------------------------------|-------| 
| user_messages                               | 90,439 | 
| blocks_ranges                               | 23,307 | 
| contacts                                    | 17,074 | 
| bundles                                     | 14,999 | 
| push_notification_client_queries            | 13,549 | 
| installations                               | 10,165 | 
| emoji_reactions                             | 9,240  | 
| mailserver_request_gaps                     | 3,850  | 
| keys                                        | 1,573  | 
| raw_messages                                | 1,445  | 
| mvds_messages                               | 897   | 
| sessions                                    | 414   | 
| ratchet_info_v2                             | 414   | 
| waku_keys                                   | 393   | 
| mailserver_topics                           | 363   | 
| mailserver_chat_request_ranges              | 324   | 
| chats                                       | 310   | 
| secret_installation_ids                     | 273   | 
| blocks                                      | 239   | 
| transfers                                   | 235   | 
| push_notification_client_tracked_messages   | 211   | 
| secrets                                     | 117   | 
| push_notification_client_sent_notifications | 80    | 
| mvds_states                                 | 77    | 
| browsers_history                            | 72    | 
| push_notification_client_info               | 69    | 
| whisper_keys                                | 67    | 
| mvds_peers                                  | 62    | 
| browsers                                    | 26    | 
| accounts                                    | 8     | 
| permissions                                 | 8     | 
| dapps                                       | 7     | 
| tokens                                      | 3     | 
| status_go_schema_migrations                 | 1     | 
| settings                                    | 1     | 
| mvds_schema_migrations                      | 1     | 
| mvds_epoch                                  | 1     | 
| status_protocol_go_schema_migrations        | 1     | 
| contact_code_config                         | 1     | 
| installation_metadata                       | 1     | 
| group_chat_invitations                      | 1     | 
| mailservers                                 | 0     | 
| messenger_transactions_to_validate          | 0     | 
| push_notification_server_registrations      | 0     | 
| push_notification_server_identity           | 0     | 
| push_notification_client_servers            | 0     | 
| push_notification_client_registrations      | 0     | 
| push_notification_server_notifications      | 0     | 
| pending_transactions                        | 0     | 
| favourites                                  | 0     | 
| local_notifications_preferences             | 0     | 
| bookmarks                                   | 0     | 
| chat_identity_last_published                | 0     | 
| chat_identity_contacts                      | 0     | 

```
*** Page counts for all tables with their indices *****************************

USER_MESSAGES..................................... 107178      90.9% 
CONTACTS.......................................... 3465         2.9% 
RAW_MESSAGES...................................... 1293         1.1% 
EMOJI_REACTIONS................................... 1140         0.97% 
MVDS_MESSAGES..................................... 916          0.78% 
TRANSFERS......................................... 741          0.63% 
CHATS............................................. 735          0.62% 
BUNDLES........................................... 613          0.52% 
PUSH_NOTIFICATION_CLIENT_QUERIES.................. 541          0.46% 
INSTALLATIONS..................................... 419          0.36% 
MAILSERVER_REQUEST_GAPS........................... 305          0.26% 
BLOCKS_RANGES..................................... 206          0.17% 
KEYS.............................................. 77           0.065% 
SESSIONS.......................................... 40           0.034% 
RATCHET_INFO_V2................................... 31           0.026% 
MAILSERVER_TOPICS................................. 15           0.013% 
SECRET_INSTALLATION_IDS........................... 14           0.012% 
WAKU_KEYS......................................... 14           0.012% 
PUSH_NOTIFICATION_CLIENT_SENT_NOTIFICATIONS....... 12           0.010% 
MAILSERVER_CHAT_REQUEST_RANGES.................... 11           0.009% 
PUSH_NOTIFICATION_CLIENT_TRACKED_MESSAGES......... 11           0.009% 
BLOCKS............................................ 10           0.008% 
PUSH_NOTIFICATION_CLIENT_INFO..................... 10           0.008% 
SETTINGS.......................................... 10           0.008% 
```

db file is ~500Mb

**Comment by flexsurfer** [comment]
| Table                                       | Rows  | Delta |
|---------------------------------------------|-------| -------|
| user_messages                               | 105,900 | +15,461
| blocks_ranges                               | **7** |
| contacts                                    | 20,500 | +3,426
| bundles                                     | 21,175 | +6,175
| push_notification_client_queries            | 16,101 | +2,552 
| installations                               | 14,150 | +4,000
| emoji_reactions                             | 12,000  | +3,000
| mailserver_request_gaps                     | 3,980  | +140


db file is ~520Mb

**Comment by flexsurfer** [comment]
after group chats spam

| Table                                       | Rows  | Delta |
|---------------------------------------------|-------| -------|
| user_messages                               | 122,200 | +17,000
| blocks_ranges                               | **7** |
| contacts                                    | 23,500 | +3,000
| bundles                                     | 26,175 | +5,175
| push_notification_client_queries            | 18,101 | +2,552 
| installations                               | 17,150 | +3,000
| emoji_reactions                             | 12,700  | +700
| mailserver_request_gaps                     | 4,380  | +540
| chats                     | 963  | -


db file is ~566Mb

**Comment by churik** [comment]
@flexsurfer still relevant?

**Comment by flexsurfer** [comment]
yes

**Comment by cammellos** [comment]
I'll move this to a discussion, and we can split off issues when ready to be worked on
